torch/_prims/__init__.py

```diff
@@ -684,6 +692,18 @@ def _fill_aten(a: Tensor, value: NumberType) -> Tensor:
     type_promotion=ELEMENTWISE_PRIM_TYPE_PROMOTION_KIND.DEFAULT,
 )
 
+imag = _make_prim(
+    schema="imag(Tensor self) -> Tensor",
+    meta=partial(
```

Comment:
I would just wrap the partial in another function that verifies the tensor is a complex dtype instead of modifying `_elementwise_meta` itself

Reply:
(or just leave the prim usage unchecked, that'd be even simpler)

Reply:
I added a `_complex_only_elementwise_meta`